### PR TITLE
Update friendship evo threshold to gen 8+ with config

### PR DESCRIPTION
**Pull request by kittenchilly**
## Description
Updates the friendship evo threshold to 160 like it is in gen 8+, and moves the variable so its defined in a config for easy editing.

## **Discord contact info**
kittenchilly

### CODE REVIEW

File: include/config/pokemon.h [hunk 1]
```diff
@@ -7,6 +7,9 @@
 #define P_UPDATED_ABILITIES         GEN_LATEST  // Since Gen 6, certain Pokémon have their abilities changed.
 #define P_UPDATED_EGG_GROUPS        GEN_LATEST  // Since Gen 8, certain Pokémon have gained new egg groups.
 
+// Evolution settings
+#define P_FRIENDSHIP_EVO_THRESHOLD   160  // Since Gen 8, Pokémon that evolve by friendship evolve at or above 160 friendship instead of 220.
```
Comment (AsparagusEduardo) [inline include/config/pokemon.h:11]: Most configs use the `GEN_x` system, so let's stick with that.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: include/config/pokemon.h (hunk 1 above)
Comment (kittenchilly) [reply include/config/pokemon.h:11]: Not necessarily. I think this is a good config to have fine tuned control over, and it makes code cleaner.

Comment (AsparagusEduardo) [reply include/config/pokemon.h:11]: I think someone else should weight in

Comment (AlexOn1ine) [reply include/config/pokemon.h:11]: I agree with a `GEN_x` config. There is a config for gems and type boosting items and I would treat this the same. People can still fine tune regardless.

Comment (kittenchilly) [reply include/config/pokemon.h:11]: Alright, I'll change it then

**Comment by kittenchilly** [comment]
Reviews applied

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**